scrapy/settings/__init__.py

```diff
@@ -197,6 +198,39 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def getdictorlist(self, name, default=None):
+        """Get a setting value as either an ``OrderedDict`` or a list.
+
+        If the setting is already a dict or a list, a copy of it will be
+        returned.
+
+        If it is a string it will be evaluated as JSON, or as a comma-separated
+        list of strings as a fallback.
+
+        For example, settings populated through environment variables will
```

Comment:
Do we support populating settings through environment variables? I recall there was something about it, but not sure it is documented.